src/main/java/io/confluent/connect/hdfs/TopicPartitionWriter.java

```diff
@@ -165,6 +171,15 @@ public TopicPartitionWriter(
       this.hiveUpdateFutures = hiveUpdateFutures;
       hivePartitions = new HashSet<>();
     }
+
+    String timeZoneString = (String) connectorConfig.getString(HdfsSinkConnectorConfig.TIMEZONE_CONFIG);
```

Comment:
Should we throw the exception only when time config is definitely required. We can check that if the scheduled file rotate is enabled to determine whether we need the timezone config or not. Another option is to introduce another configuration indicating that server timezone should be used instead of  UTC.